core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -294,8 +301,12 @@ public long getDefaultBlockSize() {
               info.getBlockInfo().getLength()));
         }
       });
-      BlockLocation[] ret = new BlockLocation[blockLocations.size()];
-      return blockLocations.toArray(ret);
+      BlockLocation[] ret = blockLocations.toArray(new BlockLocation[0]);
```

Comment:
Init the array to `blockLocations.length`?

Reply:
Should we sort `blockLocations` explicitly?

Reply:
Updated to preallocate the array.

Regarding sorting, the original block info list from master is already in the correct order because we add each block to file at the the time of creation. I added a unit test to verify the order. When the assumption no longer holds we can add the sort.